From 82d38f929882a9c62289b179c6fe41efed249987 Mon Sep 17 00:00:00 2001
From: yodalee <lc85301@gmail.com>
Date: Sat, 26 May 2018 01:35:44 +0800
Subject: [PATCH] add ZeroQueryDict::iterator::operator-- for gcc8

---
 src/prediction/zero_query_dict.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/src/prediction/zero_query_dict.h b/src/prediction/zero_query_dict.h
index c11aa24d..6e24b0bf 100644
--- a/src/prediction/zero_query_dict.h
+++ b/src/prediction/zero_query_dict.h
@@ -147,6 +147,17 @@ class ZeroQueryDict {
       return iter;
     }
 
+    iterator &operator--() {
+      ptr_ -= kTokenByteSize;
+      return *this;
+    }
+
+    iterator operator--(int) {
+      const iterator tmp(ptr_, string_array_);
+      ptr_ -= kTokenByteSize;
+      return tmp;
+    }
+
     iterator &operator-=(ptrdiff_t n) {
       ptr_ -= n * kTokenByteSize;
       return *this;
