diff --git a/im/pinyin/pinyin.cpp b/im/pinyin/pinyin.cpp
index 02c7c8c..b1fac2c 100644
--- a/im/pinyin/pinyin.cpp
+++ b/im/pinyin/pinyin.cpp
@@ -627,14 +627,14 @@ PinyinEngine::PinyinEngine(Instance *instance)
     predictionAction_.setShortText(_("Prediction"));
     predictionAction_.setLongText(_("Show prediction words"));
     predictionAction_.setIcon(*config_.predictionEnabled
-                                  ? "fcitx-remind-active"
-                                  : "fcitx-remind-inactive");
+                                  ? "fcitx-org.fcitx_im.fcitx5.remind-active"
+                                  : "fcitx-org.fcitx_im.fcitx5.remind-inactive");
     predictionAction_.connect<SimpleAction::Activated>(
         [this](InputContext *ic) {
             config_.predictionEnabled.setValue(!(*config_.predictionEnabled));
             predictionAction_.setIcon(*config_.predictionEnabled
-                                          ? "fcitx-remind-active"
-                                          : "fcitx-remind-inactive");
+                                          ? "fcitx-org.fcitx_im.fcitx5.remind-active"
+                                          : "fcitx-org.fcitx_im.fcitx5.remind-inactive");
             predictionAction_.update(ic);
         });
     instance_->userInterfaceManager().registerAction("pinyin-prediction",
diff --git a/modules/chttrans/chttrans.cpp b/modules/chttrans/chttrans.cpp
index c90cf7e..32548d7 100644
--- a/modules/chttrans/chttrans.cpp
+++ b/modules/chttrans/chttrans.cpp
@@ -70,8 +70,8 @@ Chttrans::Chttrans(fcitx::Instance *instance) : instance_(instance) {
                     notifications()->call<INotifications::showTip>(
                         "fcitx-chttrans-toggle",
                         _("Simplified and Traditional Chinese Translation"),
-                        tradEnabled ? "fcitx-chttrans-active"
-                                    : "fcitx-chttrans-inactive",
+                        tradEnabled ? "org.fcitx_im.fcitx5.chttrans-active"
+                                    : "org.fcitx_im.fcitx5.chttrans-inactive",
                         tradEnabled ? _("Switch to Simplified Chinese")
                                     : _("Switch to Traditional Chinese"),
                         tradEnabled ? _("Traditional Chinese is enabled.")
diff --git a/modules/chttrans/chttrans.h b/modules/chttrans/chttrans.h
index 2d9a2fa..1ef8f69 100644
--- a/modules/chttrans/chttrans.h
+++ b/modules/chttrans/chttrans.h
@@ -64,8 +64,8 @@ class Chttrans final : public fcitx::AddonInstance {
         }
         std::string icon(fcitx::InputContext *ic) const override {
             return parent_->convertType(ic) == ChttransIMType::Trad
-                       ? "fcitx-chttrans-active"
-                       : "fcitx-chttrans-inactive";
+                       ? "org.fcitx_im.fcitx5.chttrans-active"
+                       : "org.fcitx_im.fcitx5.chttrans-inactive";
         }
 
         void activate(fcitx::InputContext *ic) override {
diff --git a/modules/fullwidth/fullwidth.cpp b/modules/fullwidth/fullwidth.cpp
index 7c88842..faecd7f 100644
--- a/modules/fullwidth/fullwidth.cpp
+++ b/modules/fullwidth/fullwidth.cpp
@@ -68,8 +68,8 @@ Fullwidth::Fullwidth(Instance *instance) : instance_(instance) {
                 if (notifications()) {
                     notifications()->call<INotifications::showTip>(
                         "fcitx-fullwidth-toggle", _("Full width character"),
-                        enabled_ ? "fcitx-fullwidth-active"
-                                 : "fcitx-fullwidth-inactive",
+                        enabled_ ? "org.fcitx_im.fcitx5.fullwidth-active"
+                                 : "org.fcitx_im.fcitx5.fullwidth-inactive",
                         _("Full width Character"),
                         enabled_ ? _("Full width Character is enabled.")
                                  : _("Full width Character is disabled."),
diff --git a/modules/fullwidth/fullwidth.h b/modules/fullwidth/fullwidth.h
index 0b3f782..d2c3585 100644
--- a/modules/fullwidth/fullwidth.h
+++ b/modules/fullwidth/fullwidth.h
@@ -30,8 +30,8 @@ class Fullwidth final : public fcitx::AddonInstance {
                                      : _("Half width Character");
         }
         std::string icon(fcitx::InputContext *) const override {
-            return parent_->enabled_ ? "fcitx-fullwidth-active"
-                                     : "fcitx-fullwidth-inactive";
+            return parent_->enabled_ ? "org.fcitx_im.fcitx5.fullwidth-active"
+                                     : "org.fcitx_im.fcitx5.fullwidth-inactive";
         }
 
         void activate(fcitx::InputContext *ic) override {
diff --git a/modules/punctuation/punctuation.cpp b/modules/punctuation/punctuation.cpp
index 4b68361..1a47611 100644
--- a/modules/punctuation/punctuation.cpp
+++ b/modules/punctuation/punctuation.cpp
@@ -146,7 +146,7 @@ Punctuation::Punctuation(Instance *instance)
                 if (notifications()) {
                     notifications()->call<INotifications::showTip>(
                         "fcitx-punc-toggle", _("Punctuation"),
-                        enabled() ? "fcitx-punc-active" : "fcitx-punc-inactive",
+                        enabled() ? "org.fcitx_im.fcitx5.punc-active" : "org.fcitx_im.fcitx5.punc-inactive",
                         _("Punctuation"),
                         enabled() ? _("Full width punctuation is enabled.")
                                   : _("Full width punctuation is disabled."),
diff --git a/modules/punctuation/punctuation.h b/modules/punctuation/punctuation.h
index 0a162a5..445d222 100644
--- a/modules/punctuation/punctuation.h
+++ b/modules/punctuation/punctuation.h
@@ -58,8 +58,8 @@ class Punctuation final : public fcitx::AddonInstance {
                                       : _("Half width punctuation");
         }
         std::string icon(fcitx::InputContext *) const override {
-            return parent_->enabled() ? "fcitx-punc-active"
-                                      : "fcitx-punc-inactive";
+            return parent_->enabled() ? "org.fcitx_im.fcitx5.punc-active"
+                                      : "org.fcitx_im.fcitx5.punc-inactive";
         }
 
         void activate(fcitx::InputContext *ic) override {
