From 6f779c832a1b73d5735bafebb5782c290ced52ed Mon Sep 17 00:00:00 2001
From: tinywrkb <tinywrkb@gmail.com>
Date: Sat, 8 May 2021 22:43:31 +0300
Subject: [PATCH] deb: use xdg dirs instead of localstatedir

---
 src/checkers/debianrepochecker.py | 11 ++++++++---
 1 file changed, 8 insertions(+), 3 deletions(-)

diff --git a/src/checkers/debianrepochecker.py b/src/checkers/debianrepochecker.py
index 7ff8572..b7e6c4f 100644
--- a/src/checkers/debianrepochecker.py
+++ b/src/checkers/debianrepochecker.py
@@ -156,7 +156,7 @@ class DebianRepoChecker(Checker):
                 os.makedirs(os.path.join(root, path), exist_ok=True)
 
             # Create sources.list
-            sources_list = os.path.join(root, "etc/apt/sources.list")
+            sources_list = os.path.join(os.getenv('XDG_CONFIG_HOME'), 'apt/sources.list')
             with open(sources_list, "w") as f:
                 # FIXME: import GPG key, remove 'trusted=yes' which skips GPG
                 # verification
@@ -168,14 +168,19 @@ class DebianRepoChecker(Checker):
                     )
 
             # Create empty dpkg status
-            dpkg_status = os.path.join(root, "var/lib/dpkg/status")
+            dpkg_status = os.path.join(os.getenv('XDG_DATA_HOME'), 'dpkg/status')
             with open(dpkg_status, "w") as f:
                 pass
 
             # Setup generic configuration
             apt_pkg.init()
             apt_pkg.config.set("Dir", root)
-            apt_pkg.config.set("Dir::State::status", dpkg_status)
+            apt_pkg.config.set("Dir::State::status", os.path.join(os.getenv('XDG_DATA_HOME'), 'dpkg/status'))
+            apt_pkg.config.set("Dir::State", os.path.join(os.getenv('XDG_DATA_HOME'), 'apt'))
+            apt_pkg.config.set("Dir::Cache", os.path.join(os.getenv('XDG_CACHE_HOME'), 'apt'))
+            apt_pkg.config.set("Dir::Etc::SourceList", os.path.join(os.getenv('XDG_CONFIG_HOME'), 'apt/sources.list'))
+            apt_pkg.config.set("Dir::Etc::Trusted", os.path.join(os.getenv('XDG_CONFIG_HOME'), 'apt/trusted.gpg'))
+            apt_pkg.config.set("Dir::Etc::TrustedParts", os.path.join(os.getenv('XDG_CONFIG_HOME'), 'apt/trusted.gpg.d'))
             apt_pkg.config.set("Acquire::Languages", "none")
             progress = LoggerAcquireProgress(LOG)
 
-- 
2.31.1

