From 10c9c02e1ce942a8e80d8e1b57fe600950adac66 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nils=20Tonn=C3=A4tt?= <nils.tonnaett@posteo.de>
Date: Sun, 18 Apr 2021 14:21:43 +0200
Subject: [PATCH 1/4] Add feature option for installing jack development files

---
 meson_options.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/meson_options.txt b/meson_options.txt
index ea18a2ece..66791f3aa 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -54,6 +54,10 @@ option('pipewire-jack',
        description: 'Enable pipewire-jack integration',
        type: 'feature',
        value: 'auto')
+option('jack-devel',
+       description: 'Install jack development files',
+       type: 'feature',
+       value: 'disabled')
 option('libjack-path',
        description: 'Where to install the libjack.so library',
        type: 'string')
-- 
GitLab


From 75ff786fe66b5fd74dc2369b2869ae1c59ded169 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nils=20Tonn=C3=A4tt?= <nils.tonnaett@posteo.de>
Date: Sun, 18 Apr 2021 15:14:30 +0200
Subject: [PATCH 2/4] Install jack headers

---
 pipewire-jack/meson.build | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pipewire-jack/meson.build b/pipewire-jack/meson.build
index 5968c84d7..c295e45e2 100644
--- a/pipewire-jack/meson.build
+++ b/pipewire-jack/meson.build
@@ -1,2 +1,5 @@
 jack_inc = include_directories('.')
+if get_option('jack-devel').enabled()
+  install_subdir('jack', install_dir: get_option('includedir'), strip_directory: false)
+endif
 subdir('src')
-- 
GitLab


From e3066b2dbeccaa3806a5a1ed8e96f99f28bc9e92 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nils=20Tonn=C3=A4tt?= <nils.tonnaett@posteo.de>
Date: Sun, 18 Apr 2021 15:47:58 +0200
Subject: [PATCH 3/4] Generate JACK pkgconfig file

---
 pipewire-jack/src/meson.build | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/pipewire-jack/src/meson.build b/pipewire-jack/src/meson.build
index 148db2829..32d37c484 100644
--- a/pipewire-jack/src/meson.build
+++ b/pipewire-jack/src/meson.build
@@ -47,6 +47,15 @@ pipewire_jack = shared_library('jack',
     install_dir : libjack_path,
 )
 
+if get_option('jack-devel').enabled()
+  pkgconfig.generate(filebase : 'jack',
+  libraries : pipewire_jack,
+  name : 'jack',
+  description : 'PipeWire JACK API',
+  version : '1.9.17',
+  extra_cflags : '-D_REENTRANT')
+endif
+
 pipewire_jackserver = shared_library('jackserver',
     pipewire_jackserver_sources,
     soversion : soversion,
-- 
GitLab


From 9fc743c94592b692221776c11dccccec4e5613ee Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Nils=20Tonn=C3=A4tt?= <nils.tonnaett@posteo.de>
Date: Sun, 18 Apr 2021 18:14:34 +0200
Subject: [PATCH 4/4] Add server_libs variable to jack pkgconf file

---
 pipewire-jack/src/meson.build | 19 ++++++++++---------
 1 file changed, 10 insertions(+), 9 deletions(-)

diff --git a/pipewire-jack/src/meson.build b/pipewire-jack/src/meson.build
index 32d37c484..83e340ba5 100644
--- a/pipewire-jack/src/meson.build
+++ b/pipewire-jack/src/meson.build
@@ -47,15 +47,6 @@ pipewire_jack = shared_library('jack',
     install_dir : libjack_path,
 )
 
-if get_option('jack-devel').enabled()
-  pkgconfig.generate(filebase : 'jack',
-  libraries : pipewire_jack,
-  name : 'jack',
-  description : 'PipeWire JACK API',
-  version : '1.9.17',
-  extra_cflags : '-D_REENTRANT')
-endif
-
 pipewire_jackserver = shared_library('jackserver',
     pipewire_jackserver_sources,
     soversion : soversion,
@@ -78,6 +69,16 @@ pipewire_jackserver = shared_library('jacknet',
     install_dir : libjack_path,
 )
 
+if get_option('jack-devel').enabled()
+  pkgconfig.generate(filebase : 'jack',
+  libraries : [pipewire_jack, pipewire_jackserver],
+  name : 'jack',
+  description : 'PipeWire JACK API',
+  version : '1.9.17',
+  extra_cflags : '-D_REENTRANT',
+  variables: ['server_libs=-L${libdir} -ljackserver'])
+endif
+
 if sdl_dep.found()
   executable('video-dsp-play',
     '../examples/video-dsp-play.c',
-- 
GitLab

