From 8b3210c991f08a025e489fdf628985115675d365 Mon Sep 17 00:00:00 2001
From: tinywrkb <tinywrkb@gmail.com>
Date: Mon, 18 Jul 2022 15:26:19 +0300
Subject: [PATCH] config-top.h: set devenv defaults

---
 config-top.h | 10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/config-top.h b/config-top.h
index 735f75e..c97168f 100644
--- a/config-top.h
+++ b/config-top.h
@@ -61,6 +61,7 @@
 /* #define DONT_REPORT_BROKEN_PIPE_WRITE_ERRORS */
 
 /* The default value of the PATH variable. */
+#define DEFAULT_PATH_VALUE "/usr/bin"
 #ifndef DEFAULT_PATH_VALUE
 #define DEFAULT_PATH_VALUE \
   "/usr/local/bin:/usr/local/sbin:/usr/bin:/usr/sbin:/bin:/sbin:."
@@ -72,6 +73,7 @@
 
 /* The value for PATH when invoking `command -p'.  This is only used when
    the Posix.2 confstr () function, or CS_PATH define are not present. */
+#define STANDARD_UTILS_PATH "/usr/bin"
 #ifndef STANDARD_UTILS_PATH
 #define STANDARD_UTILS_PATH \
   "/bin:/usr/bin:/sbin:/usr/sbin:/etc:/usr/etc"
@@ -88,17 +90,17 @@
 #define KSH_COMPATIBLE_SELECT
 
 /* Default interactive shell startup file. */
-#define DEFAULT_BASHRC "~/.bashrc"
+#define DEFAULT_BASHRC "~/.config/bash/bash.bashrc"
 
 /* System-wide .bashrc file for interactive shells. */
-/* #define SYS_BASHRC "/etc/bash.bashrc" */
+#define SYS_BASHRC "/usr/lib/sdk/devenv/etc/bash.bashrc"
 
 /* System-wide .bash_logout for login shells. */
-/* #define SYS_BASH_LOGOUT "/etc/bash.bash_logout" */
+#define SYS_BASH_LOGOUT "/usr/lib/sdk/devenv/etc/bash.bash_logout"
 
 /* Define this to make non-interactive shells begun with argv[0][0] == '-'
    run the startup files when not in posix mode. */
-/* #define NON_INTERACTIVE_LOGIN_SHELLS */
+#define NON_INTERACTIVE_LOGIN_SHELLS
 
 /* Define this if you want bash to try to check whether it's being run by
    sshd and source the .bashrc if so (like the rshd behavior).  This checks
-- 
2.36.1

