diff --git a/CMakeLists.txt b/CMakeLists.txt
index c17173d..23729ca 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -14,12 +14,12 @@ message(STATUS "Build configuration: " ${CMAKE_BUILD_TYPE})
 
 # detect if libstdc++ is in use to know if libstdc++fs should be linked
 include(CheckCXXSourceCompiles)
-check_cxx_source_compiles("#include <iostream>
-    #ifndef __GLIBCXX__
-    #error not using libstdc+++
-    #endif
-    int main() { return 0; }"
-    USING_LIBSTDCXX)
+#check_cxx_source_compiles("#include <iostream>
+#    #ifndef __GLIBCXX__
+#    #error not using libstdc+++
+#    #endif
+#    int main() { return 0; }"
+#    USING_LIBSTDCXX)
 
 # check requirements for std::filesystem
 if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND
diff --git a/qimgv/CMakeLists.txt b/qimgv/CMakeLists.txt
index 79eeacc..25730ef 100644
--- a/qimgv/CMakeLists.txt
+++ b/qimgv/CMakeLists.txt
@@ -23,9 +23,9 @@ set_target_properties(qimgv PROPERTIES CXX_EXTENSIONS OFF)
 
 # LINK STUFF
 target_link_libraries(qimgv PRIVATE Qt5::Core Qt5::Widgets Qt5::Concurrent)
-if(USING_LIBSTDCXX)
-    target_link_libraries(qimgv PRIVATE stdc++fs)
-endif()
+#if(USING_LIBSTDCXX)
+#    target_link_libraries(qimgv PRIVATE stdc++fs)
+#endif()
 
 # OPTION DEFINITIONS, LINKING
 if(EXIV2)
